### PR TITLE
【KernelGen】Add _index_put_impl_ operator

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
### PR Category
Operator

### Type of Change
New Feature

### Description
Add `_index_put_impl_` operator implementation with Triton kernel.

- Implementation mode: `manual_kernel`
- Accuracy test: 46/46 passed

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
**torch.bfloat16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [268435456] | 0.0201 | 0.0352 | 0.571 |
| [32, 32] | 0.0092 | 0.0074 | 1.257 |
| [1024, 1024] | 0.0096 | 0.0076 | 1.265 |
| [512, 512, 512] | 0.0097 | 0.0082 | 1.175 |

**torch.float16**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [268435456] | 0.0199 | 0.0423 | 0.471 |
| [32, 32] | 0.0087 | 0.0075 | 1.158 |
| [1024, 1024] | 0.0090 | 0.0074 | 1.207 |
| [512, 512, 512] | 0.0101 | 0.0071 | 1.434 |
| [268435456] acc=true | 0.1940 | 0.0465 | 4.171 |
| [32, 32] acc=true | 0.0809 | 0.0085 | 9.536 |
| [512, 512, 512] acc=true | 0.1201 | 0.0092 | 13.119 |

**torch.float32**

| Shape | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|-------------------:|-------------------:|--------:|
| [268435456] | 0.0210 | 0.0354 | 0.593 |
| [32, 32] | 0.0092 | 0.0080 | 1.147 |
| [1024, 1024] | 0.0087 | 0.0076 | 1.147 |
| [512, 512, 512] | 0.0092 | 0.0079 | 1.167 |
| [268435456] acc=true | 0.1943 | 0.0434 | 4.475 |
| [32, 32] acc=true | 0.0813 | 0.0073 | 11.100 |
| [512, 512, 512] acc=true | 0.1200 | 0.0078 | 15.428 |

**Overall: median speedup = 1.232x, mean speedup = 3.912x** (18 data points)

---
_Generated by auto_gen tool with Claude Code_

### CODE REVIEW

**Comment by Schopenhauer-loves-Hegel** [comment]
Closing in favor of grouped PRs to reduce review overhead.

File: tests/test_reduction_ops.py [hunk 1]
```diff
@@ -2552,3 +2552,134 @@ def test_accuracy_bincount_minlength(shape, num_classes, minlength):
...
+    accumulate = False
+    inp = torch.randn(
+        input_shape, dtype=dtype, device=flag_gems.device, requires_grad=False
+    )
+
+    indices = gen_indices_for_index_put(input_shape, indices_shape, accumulate, is_bool)
+
+    if is_bool:
+        if flag_gems.vendor_name == "tsingmicro":
+            K = indices[0].to(device="cpu").sum().item()
```
Comment (tengqm) [inline tests/test_reduction_ops.py]: Why this special test?

File: feb27_migration_todo.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Feb27 PR修复进度 (修复CI问题+代码质量review)
```
Comment (tengqm) [inline feb27_migration_todo.md]: Remove this file from PR.

**Comment by factnn** [comment]
## Unit Test Results

```
index_put_impl: 46 passed, 3007 deselected in 50.82s
```

All `_index_put_impl_` tests passed (acc_false, acc_true, unsafe_param, error_all_none).

**Comment by factnn** [comment]
## Performance Benchmark Results

```
Operator: _index_put_impl_ (acc=false, dtype=torch.float32)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.021            0.036               0.589          [268435456], idx=[65536]
SUCCESS               0.008            0.008               0.942          [32,32], idx=[8,8]
SUCCESS               0.011            0.008               1.272          [32,32], idx=[8,2x8]
SUCCESS               0.009            0.008               1.126          [32,32], idx=[2x8]
SUCCESS               0.008            0.008               0.992          [1024,1024], idx=[64,64]
SUCCESS               0.013            0.007               1.754          [1024,1024], idx=[64,4x64]
SUCCESS               0.011            0.009               1.140          [1024,1024], idx=[4x64]
SUCCESS               0.009            0.007               1.286          [512,512,512], idx=[128,128,128]
SUCCESS               0.016            0.008               2.020          [512,512,512], idx=[2x128,128,128]
SUCCESS               0.414            0.098               4.214          [512,512,512], idx=[2x128]

Operator: _index_put_impl_ (acc=true, dtype=torch.float16)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.194            0.046               4.258          [268435456], idx=[65536]
SUCCESS               0.078            0.009               8.627          [32,32], idx=[8,8]
SUCCESS               0.084            0.009               9.571          [1024,1024], idx=[64,64]
SUCCESS               0.117            0.009              12.505          [512,512,512], idx=[128,128,128]
SUCCESS               0.117            0.009              13.638          [512,512,512], idx=[2x128,2x128,2x128]

Operator: _index_put_impl_ (acc=true, dtype=torch.float32)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.117            0.008              15.095          [512,512,512], idx=[128,128,128]
SUCCESS               0.117            0.009              12.328          [512,512,512], idx=[2x128,2x128,2x128]
```

**Review by tengqm — APPROVED** [review]

File: benchmark/test_select_and_slice_perf.py [hunk 3]
```diff
@@ -710,3 +710,48 @@ def test_index_acc_perf():
...
+        else:
+            values = torch.randn(
+                values_shape, dtype=dtype, device=flag_gems.device, requires_grad=False
+            )
+        yield inp, indices, values, accumulate, unsafe
+
+    return inner
+
+
+@pytest.mark.index_put_impl
```
Comment (tengqm) [inline benchmark/test_select_and_slice_perf.py:738]: This operator name has a trailing underscore, but it is not an inplace verion of something, right?

**Review by tengqm — APPROVED** [review]

**Review by tengqm — APPROVED** [review]
